src/test/java/org/jabref/logic/importer/fetcher/ArXivTest.java

```diff
@@ -13,210 +14,296 @@
     }
 
     @Test
     void searchEntryByIdWith4DigitsAndPrefix() throws Exception {
-        assertEquals(Optional.of(sliceTheoremPaper), finder.performSearchById("arXiv:1405.2249"));
+        assertEquals(Optional.of(sliceTheoremPaper), fetcher.performSearchById("arXiv:1405.2249"));
     }
 
     @Test
     void searchEntryByIdWith4DigitsAndPrefixAndNotTrimmed() throws Exception {
-        assertEquals(Optional.of(sliceTheoremPaper), finder.performSearchById("arXiv : 1405. 2249"));
+        assertEquals(Optional.of(sliceTheoremPaper), fetcher.performSearchById("arXiv : 1405. 2249"));
     }
 
     @Test
     void searchEntryByIdWith5Digits() throws Exception {
         assertEquals(Optional.of(
                 "An Optimal Convergence Theorem for Mean Curvature Flow of Arbitrary Codimension in Hyperbolic Spaces"),
-                finder.performSearchById("1503.06747").flatMap(entry -> entry.getField(StandardField.TITLE)));
+                fetcher.performSearchById("1503.06747").flatMap(entry -> entry.getField(StandardField.TITLE)));
     }
 
     @Test
     void searchWithMalformedIdThrowsException() throws Exception {
-        assertThrows(FetcherException.class, () -> finder.performSearchById("123412345"));
+        assertThrows(FetcherException.class, () -> fetcher.performSearchById("123412345"));
     }
 
     @Test
     void searchIdentifierForSlicePaper() throws Exception {
         sliceTheoremPaper.clearField(StandardField.EPRINT);
 
-        assertEquals(ArXivIdentifier.parse("1405.2249"), finder.findIdentifier(sliceTheoremPaper));
+        assertEquals(ArXivIdentifier.parse("1405.2249"), fetcher.findIdentifier(sliceTheoremPaper));
     }
 
     @Test
     void searchEmptyId() throws Exception {
-        assertEquals(Optional.empty(), finder.performSearchById(""));
+        assertEquals(Optional.empty(), fetcher.performSearchById(""));
     }
 
     @Test
     void searchWithHttpUrl() throws Exception {
-        assertEquals(Optional.of(sliceTheoremPaper), finder.performSearchById("http://arxiv.org/abs/1405.2249"));
+        assertEquals(Optional.of(sliceTheoremPaper), fetcher.performSearchById("http://arxiv.org/abs/1405.2249"));
     }
 
     @Test
     void searchWithHttpsUrl() throws Exception {
-        assertEquals(Optional.of(sliceTheoremPaper), finder.performSearchById("https://arxiv.org/abs/1405.2249"));
+        assertEquals(Optional.of(sliceTheoremPaper), fetcher.performSearchById("https://arxiv.org/abs/1405.2249"));
     }
 
     @Test
     void searchWithHttpsUrlNotTrimmed() throws Exception {
-        assertEquals(Optional.of(sliceTheoremPaper), finder.performSearchById("https : // arxiv . org / abs / 1405 . 2249 "));
+        assertEquals(Optional.of(sliceTheoremPaper), fetcher.performSearchById("https : // arxiv . org / abs / 1405 . 2249 "));
+    }
+
+    @Test
+    @Override
+    public void supportsAuthorSearch() throws Exception {
+        List<BibEntry> results = fetcher.performSearch("au:\"Tobias Diez\"");
```

Comment:
These new tests are fine, but they are not really using the new interface that you introduced. Is there are reason for this?

Reply:
The AdvancedFetcher Interface is only used for fetchers that use URL parameters for fielded search. Since in this case arXiv uses the query string for fielded search, this is not necessary.
An example where this Interface will be used is the IEEE fetcher in the future. The IEEE Fetcher Tests that should be using the interface are not implemented yet, due to my IEEE API key not being activated until recently.

Reply:
This is contained in the JavaDoc from the interface the test implements, isn't it?

Reply:
I thought the end goal was to support more complex search queries with a syntax that is the same across all "advanced fetchers" (and parsed to the "AdvancedSearchQuery" object). Thus, I think also the arXiv fetcher should implement the `AdvancedFetcher` interface, and then internally convert it to the correct query using "au:" etc.

Reply:
The thought behind the current state is, that the fetchers that support fielded search as part of the query string do not have to be changed. We can integrate the perfromComplexSearchQuery method into the SearchBasedFetcher interface. The problem with this is, that not all APIs and their corresponding fetchers will be able to support all fields that are included in the AdvancedSearchConfig/ComplexQuery.

Reply:
End goal: right. Breaking it up in smaller PRs to ease reviewing. Maybe, it was too much to include the search config now? - Howe6, to me, that part was already finished. Thus, I thought it is a good package to review. There will be follow up PRs.

@DominikVoigt Maybe you can show one implementation this week or do the issues with keys prevent you from that?

Reply:
Solved the key issue yesterday. I'll work on one implementation this week.

Reply:
We can leave it like this for the moment, but the tests should get rewritten to use the performComplexSearch method at some point.